dlt/common/schema/schema.py

```diff
@@ -177,6 +184,71 @@ def coerce_row(self, table_name: str, parent_table: str, row: StrAny) -> Tuple[D
 
         return new_row, updated_table_partial
 
+    def resolve_evolution_settings_for_table(self, parent_table: str, table_name: str, schema_evolution_settings_override: TSchemaEvolutionSettings) -> TSchemaEvolutionModes:
+
+        def resolve_single(settings: TSchemaEvolutionSettings) -> TSchemaEvolutionModes:
+            settings = settings or {}
+            if isinstance(settings, str):
+                return TSchemaEvolutionModes(table=settings, column=settings, column_variant=settings)
+            return settings
+
+        # find table settings
+        table_with_settings = parent_table or table_name
+
+        # modes
+        table_evolution_modes = resolve_single(self.tables.get(table_with_settings, {}).get("schema_evolution_settings", {}))
+        schema_evolution_modes = resolve_single(self._settings.get("schema_evolution_settings", {}))
+        overide_modes = resolve_single(schema_evolution_settings_override)
+
+        # resolve to correct settings dict
+        settings = cast(TSchemaEvolutionModes, {**DEFAULT_SCHEMA_EVOLUTION_MODES, **schema_evolution_modes, **table_evolution_modes, **overide_modes})
+
+        return settings
+
+
+    def check_schema_update(self, parent_table: str, table_name: str, row: DictStrAny, partial_table: TPartialTableSchema, schema_evolution_settings_override: TSchemaEvolutionSettings) -> Tuple[DictStrAny, TPartialTableSchema]:
+        """Checks if schema update mode allows for the requested changes, filter row or reject update, depending on the mode"""
+
+        assert partial_table
+
+        # for now we defined the schema as new if there are no data columns defined
+        has_columns = self.has_data_columns
+        if not has_columns:
+            return row, partial_table
+
+        evolution_modes = self.resolve_evolution_settings_for_table(parent_table, table_name, schema_evolution_settings_override)
+
+        # default settings allow all evolutions, skipp all else
+        if evolution_modes == DEFAULT_SCHEMA_EVOLUTION_MODES:
+            return row, partial_table
+
+        table_exists = table_name in self.tables and len(self.tables[table_name].get("columns", {}))
+
+        # check case where we have a new table
+        if not table_exists:
+            if evolution_modes == "freeze-and-trim":
```

Comment:
huh? `evolution_modes` is dict. how you can compare that. also you are missing tests